### ISSUE TITLE
Using Reason-Apollo with Create-React-App

### ISSUE THREAD

**Issue by idkjs** [apollographql/reason-apollo#95]
**Issue Labels**

- [x] has-reproduction
- [ ] feature
- [ ] docs
- [ ] blocking
- [ ] good first issue

I took the `examples/swapi` demo and tried to run it with create-react-app`.
[reproduction](https://github.com/idkjs/reason-apollo-crra/tree/feature/with-crra)
This is the error in browser. Project compiles successfully.
![screenshot](https://raw.githubusercontent.com/idkjs/reason-apollo-crra/feature/with-crra/screenshot.png)

Any idea what is going on here?

**Comment by emmenko** [comment]
Yes, it's because of this bug #80 when you have `es6` as output target. Hopefully it will be merged soon 😇

**Comment by idkjs** [comment]
I tried the fix locally, getting same error.

**Comment by emmenko** [comment]
Have you clean and rebuild with bsb?

**Comment by emmenko** [comment]
You can try with my branch otherwise (I'm keeping track of the pending fixes).

```
"reason-apollo": "emmenko/reason-apollo#pending-patches-0.10.0"
```

**Comment by idkjs** [comment]
doesnt work for me. ran clean and `rm -rf node_modules lib .merlin` reinstalled. Same error. [reproduction](https://github.com/idkjs/reason-apollo-crra/tree/feature/with-crra) is updated.

**Comment by emmenko** [comment]
I'll take a look later at the repo then

**Comment by Gregoirevda** [comment]
@emmenko This isn't the `GraphqlTag is not a function error`, happening because of the bindings (might not be an issue with the new bs-platform), but `parse is not a function`.
`graphql-tag` is using 3 entry points:
main -> bundle
module and next js -> ES6
For some reason, reason-scripts uses ES6 version and there's an error with this import
`var parse = require("graphql/language/parse")` 
`require` is not defined which leads to an undefined `parse`

**Comment by Gregoirevda** [comment]
as references https://github.com/graphql/graphql-js/issues/1248 and https://github.com/facebook/create-react-app/pull/4085

**Comment by br1anchen** [comment]
@Gregoirevda did some investigation about `CRA`(create-react-app) with `react-apollo` and `reason-apollo`.
`CRA` (`react-script` v1.1.5) works fine with `react-apollo`(v2.0.4) which requires `graphql-tag`(v^2.9.2) (`graphql`).
So I think the problem is how `bs-loader` loads `reason-apollo` in this case.
Checked `package-specs` of `reason-apollo`:
```
"package-specs": [
    {
      "module": "es6",
      "in-source": true
    }
  ],
```
I suspect that's the reason, `reason-script` uses `ES6` module for `reason-apollo`.
However, [`bs-loader`](https://github.com/reasonml-old/bs-loader) recommends 
using `bsb` in-source builds, so by set `inSource: true` as default `bs-loader` options in `reason-script` seems fixing the problem.
I made [PR](https://github.com/reasonml-old/reason-scripts/pull/77) for it, even though `reason-script` will not be [actively](https://github.com/reasonml-old/reason-scripts/issues/69#issuecomment-422896414) maintained.
Feel free to fork/tag my forked [repo](https://github.com/br1anchen/reason-scripts) in your `package.json` to try out.
